### ISSUE TITLE
wrong unused_optional_binding behavior in guard+try

### ISSUE THREAD

**Issue by Sega-Zero** [realm/SwiftLint#1447]
Having that kind of code:
```swift
guard let _ = try? self.updateInSecureStore() else {
    print("couldn't replace the value in keychain")
    return
}
```

It is 100% correct, but swiftlint treats it as a warning. So, i should change the code to that just to silence that warning?

```swift
guard let result = try? self.updateInSecureStore(), result != nil else {
    print("couldn't replace the value in keychain")
    return
}
```

I believe that first snippet should be treated as a correct code.

**Comment by marcelofabri** [comment]
Duplicate of #1432
